### ISSUE TITLE
Bug: Missing values in weight variable

### ISSUE THREAD

**Issue by nk3** [mpi2/stats_working_group#13]
Hi,
in the large scale test of PhenStat with weight we have numerous failures giving this error message  "Error in ctrfn(levels(x), contrasts = contrasts) :   contrasts not defined for 0 degrees of freedom"

The error is arising from the model fit when the model includes weight as a covariate. The error is unsurprising in that the dataset had no weight measures in the knockout mouse.   I have made some mock datasets to test when it will work and when it won't 

![snap1](https://f.cloud.github.com/assets/1395098/2137869/f4400328-932c-11e3-9510-6f0ac1dc4407.jpg)

Basically it will work as long as there is a single data point in each genotype*sex group.  Personally I think only having one data point in each group is very low and means the estimates for weight will be poor.  I think in this situation weight as a covariate will not be significant as the errors estimates will be so large and it will default to without weight.  

so this suggests we just need checking that there are weight measures in each genotype sex combination and if not revert to without weight. 

BW,
Natasha
